admin/jobs/river/validate_deployments.go

```diff
@@ -27,8 +28,16 @@ func (w *ValidateDeploymentsWorker) Work(ctx context.Context, job *river.Job[Val
 const validateDeploymentsForProjectTimeout = 5 * time.Minute
 
 func (w *ValidateDeploymentsWorker) validateDeployments(ctx context.Context) error {
-	// Iterate over batches of projects
+	// Resolve batch size from config
 	limit := 100
+	if w.admin.ProvisionerMaxConcurrency <= 100 {
+		limit = w.admin.ProvisionerMaxConcurrency
+	} else {
+		w.admin.Logger.Warn("validate deployments: provisioner max concurrency set too high, using maximum concurrency of 100", zap.Int("provisioner_max_concurrency", w.admin.ProvisionerMaxConcurrency), observability.ZapCtx(ctx))
+	}
```

Comment:
Is a hard cap necessary or can it just be:
```suggestion
	limit := 100
	if w.admin.ProvisionerMaxConcurrency > 0 {
		limit = w.admin.ProvisionerMaxConcurrency
	}
```